src/framework/matrix.hpp

```diff
@@ -35,6 +35,8 @@ Multiplication is done with the C wrapper of the fortran blas library.
 #include <iostream>
 #include <vector>
 #include <array>
+#include "framework/linalg/enable_if_numeric.hpp"
+#include "framework/linalg/almost_equal.hpp"
```

Comment:
Are dependencies are in danger of getting a bit circular here since matrix is one of the primitive classes imported included in a lot of places. I'm hesitant to add methods to the matrix class that require an import until we do a proper refactor.

Reply:
Yeah, Agree., but that's why we put the include guards at the beginning of every header file, to avoid recursions.